golem-worker-service-base/src/service/template/default.rs

```diff
@@ -10,86 +11,65 @@ use golem_common::config::RetryConfig;
 use golem_common::model::TemplateId;
 use golem_common::retries::with_retries;
 use golem_service_base::model::Template;
-use golem_service_base::service::auth::{AuthService, Permission, WithAuth, WithNamespace};
 use http::Uri;
-use std::sync::Arc;
 use tracing::info;
 
-pub type TemplateResult<T, Namespace> = Result<WithNamespace<T, Namespace>, TemplateServiceError>;
+pub type TemplateResult<T> = Result<T, TemplateServiceError>;
 
 #[async_trait]
-pub trait TemplateService<AuthCtx, Namespace> {
+pub trait TemplateService<AuthCtx> {
     async fn get_by_version(
         &self,
         template_id: &TemplateId,
         version: i32,
         auth_ctx: &AuthCtx,
-    ) -> TemplateResult<Template, Namespace>;
+    ) -> TemplateResult<Template>;
 
     async fn get_latest(
         &self,
         template_id: &TemplateId,
         auth_ctx: &AuthCtx,
-    ) -> TemplateResult<Template, Namespace>;
+    ) -> TemplateResult<Template>;
 }
 
 #[derive(Clone)]
-pub struct TemplateServiceDefault<AuthCtx, Namespace> {
+pub struct RemoteTemplateService {
     uri: Uri,
     retry_config: RetryConfig,
-    auth_service: InnerAuthService<AuthCtx, Namespace>,
 }
 
-type InnerAuthService<AuthCtx, Namespace> =
-    Arc<dyn AuthService<WithAuth<TemplateId, AuthCtx>, Namespace> + Send + Sync>;
-
-impl<AuthCtx, Namespace> TemplateServiceDefault<AuthCtx, Namespace> {
-    pub fn new(
-        uri: Uri,
-        retry_config: RetryConfig,
-        auth_service: InnerAuthService<AuthCtx, Namespace>,
-    ) -> Self {
-        Self {
-            uri,
-            retry_config,
-            auth_service,
-        }
+impl RemoteTemplateService {
+    pub fn new(uri: Uri, retry_config: RetryConfig) -> Self {
+        Self { uri, retry_config }
     }
 }
 
 #[async_trait]
-impl<AuthCtx, Namespace> TemplateService<AuthCtx, Namespace>
-    for TemplateServiceDefault<AuthCtx, Namespace>
+impl<AuthCtx> TemplateService<AuthCtx> for RemoteTemplateService
 where
-    Namespace: Send + Sync,
-    AuthCtx: Clone + Send + Sync,
+    AuthCtx: IntoIterator<Item = (String, String)> + Clone + Send + Sync,
```

Comment:
For RemoteTemplateService, which will use grpc, I added a constraint to create a Metadata map, which will add entries to the grpc request metadata using `with_metadata`

In cloud we can put headers and such here.